### PR TITLE
libfs: support FileInfo methods in the billy FS layer

### PR DESCRIPTION
**Pull request by strib**
Depends on #1108.

Issue: KBFS-2338

### CODE REVIEW

File: kbfsgit/fs.go [hunk 1]
```diff
@@ -213,8 +243,36 @@ func (fs *FS) TempFile(dir, prefix string) (billy.File, error) {
...
+	if err != nil {
+		return nil, err
+	}
+
+	fis = make([]os.FileInfo, 0, len(children))
+	for name, ei := range children {
+		fis = append(fis, &FileInfo{
+			fs:       fs,
+			ei:       ei,
+			fullpath: path + "/" + name,
```
Comment (songgao) [inline kbfsgit/fs.go]: Maybe use `path.Join` here?

File: kbfsgit/file_info.go [hunk 2]
```diff
@@ -7,41 +7,63 @@ package kbfsgit
...
+	ei       libkbfs.EntryInfo
+	fullpath string
 }
 
 var _ os.FileInfo = (*FileInfo)(nil)
 
 // Name implements the os.FileInfo interface for FileInfo.
 func (fi *FileInfo) Name() string {
-	return ""
+	return fi.fullpath
```
Comment (songgao) [inline kbfsgit/file_info.go]: I think this is supposed to be base name so `path.Base(fi.fullpath)`?

File: kbfsgit/fs.go [hunk 3]
```diff
@@ -188,8 +199,27 @@ func (fs *FS) Open(filename string) (billy.File, error) {
...
+
+	_, ei, err := fs.config.KBFSOps().Lookup(fs.ctx, n, base)
+	if err != nil {
+		return nil, err
+	}
+
+	return &FileInfo{
+		fs:       fs,
+		ei:       ei,
+		fullpath: filename,
```
Comment (songgao) [inline kbfsgit/fs.go]: Did you intend to have `fullpath` always include full path or does it not matter? It seems it's only used by logging though.

File: kbfsgit/file_info.go (hunk 2 above)
Comment (strib) [inline kbfsgit/file_info.go]: The billy comments are pretty clear that it should be the path that's passed to `Open`, which in billy-land is the full path: https://github.com/src-d/go-billy/blob/d0dc16de10d86398cc9aba4513a2453b401eceb3/fs.go#L131

File: kbfsgit/fs.go (hunk 3 above)
Comment (strib) [inline kbfsgit/fs.go]: As mentioned in the other comment, this needs to be the full path passed to `Open`.  It's used in the `FileInfo.Name()` implementation.

File: kbfsgit/file_info.go (hunk 2 above)
Comment (songgao) [inline kbfsgit/file_info.go]: Oh I see; thanks!

Comment (songgao) [inline kbfsgit/file_info.go]: Actually, that's the comment for `File.Name`, not `FileInfo.Name`. I think `os.FileInfo` interface still require this to be the base name?

Comment (strib) [inline kbfsgit/file_info.go]: Oh, d'oh, you're right.  I'll change it.

File: libfs/fs.go [hunk 4]
```diff
@@ -125,31 +127,40 @@ func (fs *FS) lookupOrCreateFile(
...
-	f billy.File, err error) {
-	fs.log.CDebugf(
-		fs.ctx, "OpenFile %s, flag=%d, perm=%o", filename, flag, perm)
-	defer func() {
-		fs.deferLog.CDebugf(fs.ctx, "OpenFile done: %+v", err)
-	}()
-
+func (fs *FS) lookupParent(filename string) (
+	parent libkbfs.Node, base string, err error) {
 	parts := strings.Split(filename, "/")
```
Comment (jzila) [inline libfs/fs.go]: Not sure if I've already suggested this for this instance, but `filepath.ToSlash(filename)` before splitting.

**Pull request merged**